dhcpv4/server4/conn.go

```diff
@@ -38,8 +38,10 @@ func NewIPv4UDPConn(iface string, port int) (*net.UDPConn, error) {
 		}
 	}
 	// Bind to the port.
-	if err := unix.Bind(fd, &unix.SockaddrInet4{Port: port}); err != nil {
-		return nil, fmt.Errorf("cannot bind to port %d: %v", port, err)
+	saddr := unix.SockaddrInet4{Port: addr.Port}
+	copy(saddr.Addr[:], addr.IP.To4())
```

Comment:
If addr.IP != nil?

Reply:
That's fine, addr.IP.To4() will be nil in that case, and nothing will be copied; which is what we want anyway: https://play.golang.org/p/0RRxwSbL1WL